tasks/run_agent.rb

```diff
@@ -0,0 +1,160 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+require 'open3'
+require 'etc'
+require 'json'
+require 'yaml'
+require_relative '../../ruby_task_helper/files/task_helper.rb'
+
+class RunAgentTask < TaskHelper
+  def task(puppet_bin: '/opt/puppetlabs/bin/puppet', wait_time: 300, _noop: false, **_kwargs)
+    @puppet_bin = puppet_bin
+    @wait_time = wait_time
+    @noop = _noop
+
+    check_running_as_root
+    check_agent_not_disabled
+    wait_for_puppet_lockfile
+    run_puppet
+  end
+
+  def wait_for_puppet_lockfile
+    waited = 0
+    while agent_locked? && waited < @wait_time
+      sleep 1
+      waited += 1
+    end
+
+    if agent_locked?
+      raise TaskHelper::Error.new(
+        'Lockfile still exists after waiting',
+        'theforeman-puppet/lockfile_timeout_expired'
+      )
+    end
+
+    waited
```

Comment:
I don't think I'm actually using this returned value at the moment.  I suppose I could bung it into the response??